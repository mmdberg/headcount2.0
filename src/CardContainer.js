import React from 'react';
import Card from './Card';
import './styles/cardContainer.css';
import PropTypes from 'prop-types';

export const CardContainer = 
  ({stats, selectCard, deselectCard, selectedCards}) => {

    const cardInfo = stats.map(district => {
      let result = 'card';
    
      selectedCards.forEach(card =>  {
        if ( district.location === card.location) {
          result = 'card clicked';
        } 
      });


      return (
        <Card location={district.location} 
          stats={district.stats} 
          key={district.location} 
          selectCard={selectCard}
          deselectCard={deselectCard}
          className={result}
        />);
    });
  
    return (
      <div className="card-container">
        {cardInfo}
      </div>
    );
  };

CardContainer.propTypes = {
  stats: PropTypes.array,
  selectCard: PropTypes.func,
  deselectCard: PropTypes.func,
  selectedCards: PropTypes.array
};