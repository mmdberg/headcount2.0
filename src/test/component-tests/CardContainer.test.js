import React from 'react';
import { shallow, mount } from 'enzyme';
import mockCardContainerData from '../../data/mock_card_container_data'
import { CardContainer } from '../../CardContainer';

describe('CardContainer', () => {
  
  it('should match the snapshot', () => {
    let wrapper = shallow(<CardContainer  data={mockCardContainerData} 
                                          selectedCards={mockCardContainerData}
                          />)

    expect(wrapper).toMatchSnapshot();
  });

});