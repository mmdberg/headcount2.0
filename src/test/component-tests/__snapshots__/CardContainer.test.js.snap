// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    data={
        Array [
            2004,
            2005,
            2006,
            2007,
            2008,
            2009,
            2010,
            2011,
            2012,
            2013,
            2014,
          ]
    }
    selectedCards={
        Array [
            2004,
            2005,
            2006,
            2007,
            2008,
            2009,
            2010,
            2011,
            2012,
            2013,
            2014,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
        <Card
          className="card clicked"
          data={undefined}
          deselectCard={undefined}
          location={undefined}
          selectCard={undefined}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "data": undefined,
          "deselectCard": undefined,
          "location": undefined,
          "selectCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
          <Card
            className="card clicked"
            data={undefined}
            deselectCard={undefined}
            location={undefined}
            selectCard={undefined}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "data": undefined,
            "deselectCard": undefined,
            "location": undefined,
            "selectCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
